app/lib/features/settings/pages/calendar_settings_page.dart

```diff
@@ -37,18 +39,39 @@ class CalendarSettingsPage extends ConsumerWidget {
                           true),
                   onToggle: (newVal) async {
                     if (newVal) {
-                      final hasPermission = await handleCalendarPermission( context, );
-                      if (!hasPermission || !context.mounted) {
-                        return;
+                      final hasPermission = await isShowCalendarPermissionInfoPage();
+                      if (hasPermission) {
+                        if (context.mounted) {
+                          final granted = await showDialog<bool>(
+                            context: context,
+                            barrierDismissible: false,
+                            builder: (BuildContext context) {
+                              return Dialog.fullscreen(
+                                child: const CalendarSyncPermissionWidget(),
+                              );
+                            },
+                          );
+
+                          if (granted == true && context.mounted) {
+                            await initCalendarSync(ignoreRejection: true);
+                            EasyLoading.showToast('Acter Calendars synced');
+                            ref.read(isCalendarSyncActiveProvider.notifier).set(true);
+                          } else {
+                            // If permission not granted, keep toggle off
+                            ref.read(isCalendarSyncActiveProvider.notifier).set(false);
+                          }
+                        }
+                      }else{
```

Comment:
```suggestion
                      } else {
```

your editor/IDE needs to be setup to autoformat dart files.

Reply:
ok